server/src/controller.py

```diff
@@ -23,8 +22,42 @@ def get_slack_users(request=None):
                 formatted_slack_users.append(formatted_slack_user)
         return json(ok(formatted_slack_users))
 
-
-def authenticate_user(request, db_session):
+def post_verify_slack_email(request=None, db_session=None):
+    slack_users_response = rest_service.slack_users().json()
+    if not slack_users_response['ok']:
+        return json(error('Slack API error'), status=504)
+    slack_user_to_email = None
+    for slack_user in slack_users_response['members']:
+        if not slack_user['deleted'] and \
+                slack_user['name'] not in config['blacklisted_slack_users'] and \
+                request.json.get('username') == slack_user['name']:
+            slack_user_to_email = slack_user
+            break
+
+    # if already_registered:
+    #       if password is correct:
+    #           log user in
+    #       else:
+    #           don't log user in, prompt to reset password
+    # else:
+    #      if password is not strong:
+    #           return for stronger password
+    #      else:
+    #           create user, ask for email confirmation
+
+    # TODO: Should write a user to db here as an attempted registration
+    # - username: VARCHAR
+    # - SALTED ENCRYPTED PASSWORD: VARCHAR?
+    # - registered: FALSE
+    # - last_attempted_registration: CURRENT DATE
+    # - confirmation_code: VARCHAR (random string)
+
+    # TODO: Make sure that the password is long enough
+    # If so, return here, front-end should refuse registration
+
+    return json(ok(request.json))
+
+def authenticate_user(request=None, db_session=None):
```

Comment:
This method won't work if request and db_session are None; they're not really optional and shouldn't have defaults imo

Reply:
I agree with Arielle